lib/remote/apilistener-configsync.cpp

```diff
@@ -393,6 +395,48 @@ void ApiListener::UpdateConfigObject(const ConfigObject::Ptr& object, const Mess
 	}
 }
 
+/**
+ * Syncs the specified object and its direct and indirect parents to the provided client
+ * in topological order of their dependency graph recursively.
+ *
+ * Objects that the client does not have access to are skipped without going through their dependency graph.
+ *
+ * Please do not use this method to forward remote generated cluster updates; it should only be used to
+ * send local updates to that specific non-nullptr client.
+ *
+ * @param object The config object you want to sync.
+ * @param azone The zone of the client you want to send the update to.
+ * @param client The JsonRpc client you send the update to.
+ * @param syncedObjects Used to cache the already synced objects.
+ */
+void ApiListener::UpdateConfigObjectWithParents(const ConfigObject::Ptr& object, const Zone::Ptr& azone,
+	const JsonRpcConnection::Ptr& client, std::unordered_set<ConfigObject*>& syncedObjects)
+{
+	if (syncedObjects.find(object.get()) != syncedObjects.end()) {
+		return;
+	}
+
+	/* don't sync objects for non-matching parent-child zones */
+	if (!azone->CanAccessObject(object)) {
+		return;
+	}
+	syncedObjects.emplace(object.get());
+
+	// Note, due to the confusing implementation of the DependencyGraph, the terms parent and child are flipped,
+	// thus we need to traverse the graph in the opposite direction. However, when changing the implementation
+	// of the DependencyGraph, this should be changed to `GetParents()` accordingly.
```

Comment:
I see you've resolved some discussions already. Suggestion: don't push (and trigger GHA) yet, instead have a look at #10263.

Reply:
No, I'm not going to do that! If I wanted to open yet another construction area, I would have done it already. So I decided to make only the minimal changes required for this PR, instead of going back to the seemingly never ending behaviour of stacking PRs on one another.

Reply:
Come on. I'd even rebase #10000 for you afterwards! (If you want.)